### PR TITLE
Moi 10

### PR DESCRIPTION
**Pull request by matbesancon**
start upgrading Mosek to MOI 0.10

### CODE REVIEW

**Comment by matbesancon** [comment]
@odow can you take a look?

File: test/runtests.jl [hunk 1]
```diff
@@ -68,92 +70,107 @@ end
...
 end
 
-@testset "Start" begin
-    # TODO this should be checked somewhere in MOI
-    @test MOI.supports(bridged, MOI.VariablePrimalStart(), MOI.VariableIndex)
+@testset "Conic problems" begin
+    MOIT.runtests(optimizer, config,
+        include=["conic", "SecondOrderCone", "Semidefinite", "Exponential", "PowerCone", "Cone"],
+        exclude=["Indicator", "basic", "linear"],
+    )
```
Comment (odow) [inline test/runtests.jl:88]: You should prefer to run these all in a single `MOI.Test.runtests` call, and explicitly exclude failing tests by their full name.

File: test/runtests.jl [hunk 2]
```diff
@@ -68,92 +70,107 @@ end
...
+            "BoundAlreadySet", # TODO throw error if bound already set
+            "test_model_ModelFilter_AbstractVariableAttribute",
+            "test_model_VariableName", # Mosek currently throws when setting twice, not when getting names
+            "test_model_Name_VariableName_ConstraintName",
+            "test_model_duplicate_VariableName",
+            "test_model_VariablePrimalStart", # able to set but not to get VariablePrimalStart
+            "test_objective_qp_ObjectiveFunction_zero_ofdiag", # MOI.ListOfModelAttributesSet
+            "test_objective_set_via_modify",
+            "test_quadratic_nonconvex_constraint_integration",
+            "test_solve_ObjectiveBound_MAX_SENSE_LP", # ObjectiveBound invalid
```
Comment (odow) [inline test/runtests.jl:171]: The quantity of failing tests here isn't good :(

You should try with
```julia
model = MOI.Bridges.full_bridge_optimizer(
    MOI.Utilities.CachingOptimizer(
        MOI.Utilities.UniversalFallback(MOI.Utilities.Model{Float64}()),
        Mosek.Optimizer(),
    ),
    Float64,
)
MOI.Test.runtests(model, MOI.Test.Config())
```

That should add a cache for all of the names etc.

Comment (matbesancon) [reply test/runtests.jl:162]: added a third group of tests for this

File: test/runtests.jl (hunk 1 above)
Comment (matbesancon) [reply test/runtests.jl]: done in last commits

**Pull request merged**